charts/node/templates/statefulset.yaml

```diff
@@ -288,6 +289,28 @@ spec:
           securityContext:
             runAsUser: 0
         {{- end }}
+        {{- if .Values.node.persistGeneratedNodeKey }}
+        - name: persist-generated-node-key
+          image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+          command: [ "/bin/sh" ]
+          args:
+            - -c
+            - |
+              set -eu {{ if .Values.initContainer.debug }}-x{{ end }}
+              NODE_KEY_PATH="/keystore/node-key"
+              if [ -f "${NODE_KEY_PATH}" ]; then
+                echo "Node key already exists, skipping node key generation"
+              else
+                {{ $.Values.node.command }} key generate-node-key --file ${NODE_KEY_PATH} \
+                && echo "Generate node key into Keystore" \
+                || echo "Failed to insert key into Keystore."
+              fi
+              NODE_PEER_ID="$(cat ${NODE_KEY_PATH} | {{ .Values.node.command }} key inspect-node-key)"
```

Comment:
This may expose the node key in the logs. Make sure to use `key inspect-node key --file <node_key_path>`

Reply:
I think this didn't exist before, but you are right is the better option.